src/core/mod.rs

```diff
@@ -24,6 +24,8 @@ use parsec_interface::requests::Opcode;
 use parsec_interface::requests::ProviderID;
 use std::collections::HashSet;
 
+pub use ipc_handler::IpcHandler;
+
```

Comment:
Here could be defined a `Speak` trait, similar to what you currently have:
```rust
trait Speak {
   pub fn connect(&self) -> Result<Box<dyn ReadWrite>>;
}
```
Or in a `src/core/speaker` module